### DISCUSSION TITLE
Random browser issues with App colors on Wasm

### DISCUSSION THREAD

**Discussion by assassin316** [unoplatform/uno#15356] [General]
Some users are reporting a completely different look in the app, somehow the app colors are not loading properly in Wasm, not sure if it's a known issue or something I need to update in the code.

Here's a sample of my app colors (no theming yet):

![image](https://github.com/unoplatform/uno/assets/17514380/bc7a7051-6c91-49b0-8ee3-f7e933ced248)

Here's what it looks like on my end:

![image](https://github.com/unoplatform/uno/assets/17514380/0a995732-27f2-41a4-baef-ebe2cce24e50)

And here is what one user sees on their end:

![image](https://github.com/unoplatform/uno/assets/17514380/28c8ab56-dbd0-4051-80dc-27e8839730ed)

The affected user is running the latest Chrome on a Win 10 machine. This has also been reported by a friend using Chrome on a Mac.

Is there something I need to do to the app colors for proper handling across all browsers?

**Comment by jeromelaban** [comment]
This looks like a theming issue. Does your app support light/dark themes?

**Comment by assassin316** [comment]
Originally I had a flat list of colors/brushes without any theming in place, 

![image](https://github.com/unoplatform/uno/assets/17514380/570a8861-8ce4-4841-b422-80be8a441bba)

Then I made all changes to support theming using Light/Dark (copied exact same colors into both dictionaries) and used ThemeResource where needed:

![image](https://github.com/unoplatform/uno/assets/17514380/14742b0b-e95a-4067-880a-24cac42221cd)

Unfortunately got same result from the user when he tried the new version:

![image](https://github.com/unoplatform/uno/assets/17514380/e29c659d-f08f-4042-9722-8949b4c20e68)

Wondering what other hacks I can try to determine the issue.

**Reply by jeromelaban** [reply]
Very curious. Are you animating those resources in any way?

**Reply by assassin316** [reply]
No but I did notice the major cultprits of this odd behavior is my RadialBrush and LinearGradientBrush when I apply them to the Page and NavigationView of the Shell. I swapped them out for a SolidColorBrush and they work fine, still investigating and will report back with more info once I have it.

**Comment by assassin316** [comment]
Ok, figured out the issue and fixed it. It's a few things.

To begin, the following didn't work:

![image](https://github.com/unoplatform/uno/assets/17514380/b2d03ad3-7b1c-474d-90d6-dad423e20965)

What was actually happening is the Wasm version was reading the "system" theme, which in the case of some users, it was Light, and all my PC's are dark, which explains why I never saw the issue.

To fix that, it was simply:

![image](https://github.com/unoplatform/uno/assets/17514380/4dc23daa-0562-4f35-8a8b-9f3b4deda320)

Next, there is an issue with the `RadialGradientBrush`, it just never picks works with the target theme, even with my fix. I specified it as such:

![image](https://github.com/unoplatform/uno/assets/17514380/0e24887c-4d15-4b39-af07-85bbe22a39d7)

Instead of digging, I just swapped it for a `LinearGradientBrush` and that worked perfectly.

And now, the app loads as expected. 👍

Will close this discussion.
